presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestKllSketchAggregationFunction.java

```diff
@@ -126,66 +140,73 @@ public void testInvalidK()
+                {INTEGER, IntStream.iterate(0, i -> i + ThreadLocalRandom.current().nextInt()).mapToLong(x -> (long) x).boxed(),
+                        (BiConsumer<BlockBuilder, Long>) INTEGER::writeLong},
+                {BIGINT, LongStream.iterate(0, i -> i + ThreadLocalRandom.current().nextLong()).boxed(), (BiConsumer<BlockBuilder, Long>) BIGINT::writeLong},
+                {REAL, DoubleStream.iterate(0, i -> i + ThreadLocalRandom.current().nextFloat()).boxed().map(x -> (long) Float.floatToIntBits((float) (double) x)),
+                        (BiConsumer<BlockBuilder, Long>) REAL::writeLong},
+                {DOUBLE, DoubleStream.iterate(0, i -> i + ThreadLocalRandom.current().nextDouble()).boxed(), (BiConsumer<BlockBuilder, Double>) DOUBLE::writeDouble},
+                {VARCHAR, Arrays.stream("abcdefghijklmnopqrstuvwxyz".split("")).map(Slices::utf8Slice), (BiConsumer<BlockBuilder, Slice>) VARCHAR::writeSlice},
+                {BOOLEAN, IntStream.iterate(0, i -> i + 1).limit(10).mapToObj(i -> i % 2 == 0), (BiConsumer<BlockBuilder, Boolean>) BOOLEAN::writeBoolean},
+                {DATE, LongStream.iterate(0, i -> i + 1).boxed(), (BiConsumer<BlockBuilder, Long>) DATE::writeLong},
+                {TIME, LongStream.iterate(0, i -> i + 1).boxed(), (BiConsumer<BlockBuilder, Long>) TIME::writeLong},
+                {TIMESTAMP, LongStream.iterate(0, i -> i + 1).boxed(), (BiConsumer<BlockBuilder, Long>) TIMESTAMP::writeLong},
+                {TIMESTAMP_WITH_TIME_ZONE, LongStream.iterate(0, i -> i + 1).map(i -> DateTimeEncoding.packDateTimeWithZone(i, TimeZoneKey.UTC_KEY)).boxed(),
+                        (BiConsumer<BlockBuilder, Long>) TIMESTAMP_WITH_TIME_ZONE::writeLong},
+                {INTERVAL_YEAR_MONTH, IntStream.iterate(0, i -> i + 1).mapToLong(x -> (long) x).boxed(), (BiConsumer<BlockBuilder, Long>) INTERVAL_YEAR_MONTH::writeLong}
+        };
     }
 
-    @Test
-    public void testVarchar()
+    @Test(dataProvider = "testTypes")
+    public void testTypes(Type type, Stream<Object> values, BiConsumer<BlockBuilder, Object> writeBlockValue)
     {
-        String[] items = "abcdefghijklmnopqrstuvwxyz".split("");
-        BlockBuilder out = VARCHAR.createBlockBuilder(null, items.length);
-        KllItemsSketch<String> sketch = KllItemsSketch.newHeapInstance(String::compareTo, new ArrayOfStringsSerDe());
-        Arrays.stream(items).forEach(item -> {
-            VARCHAR.writeString(out, item);
-            sketch.update(item);
-        });
+        int length = 100;
+        JavaAggregationFunctionImplementation function = getFunction(type);
+        BlockBuilder out = type.createBlockBuilder(null, length);
+        KllSketchAggregationState.SketchParameters parameters = KllSketchAggregationState.getSketchParameters(type);
+        KllItemsSketch sketch = KllItemsSketch.<Object>newHeapInstance(parameters.getComparator(), parameters.getSerde());
+        List addedValues = values.limit(length)
```

Comment:
This seems a bit convoluted to me. How about we use a simple `Supplier<Object> values` and use a for-loop (`i..limit`) to iterate and build the addedValues list ?

Reply:
Thanks for the suggestion, updated